.github/workflows/tests-npm.yml

```diff
@@ -0,0 +1,69 @@
+name: NPM Tests
+on:
+  workflow_call:
+    inputs:
+      env_vars:
+        required: false
+        type: string
+        default: '{}'
+      build_tsoa:
+        required: false
+        type: boolean
+        default: false
+      db_migrate:
+        required: false
+        type: boolean
+        default: false
+      docker_compose_file:
+        required: false
+        type: string
+        default: docker-compose.yml
+      tests:
+        required: false
+        type: string
+        default: '[test:unit, test:integration]'
+jobs:
+  tests:
+    name: Run tests - ${{ matrix.command }}
+    strategy:
+      fail-fast: false
+      matrix:
+        command: ${{ github.event.inputs.tests }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set environment variables from JSON
+        env:
+          ENV_VARS_JSON: "${{ inputs.env_vars }}"
+        run: |
+          echo "Parsing JSON environment variables..."
+          echo "$ENV_VARS_JSON" | jq -r 'to_entries|map("\(.key)=\(.value|tostring)")|.[]' >> $GITHUB_ENV
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+      - name: Cache Node.js modules
+        uses: actions/cache@v4
+        with:
+          path: ~/.npm
+          key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.OS }}-node-
+            ${{ runner.OS }}-
+      - name: Install Packages
+        run: npm ci
+      - name: Build tsoa
+        run: npm run tsoa:build
+        if: inputs.build_tsoa == true
```

Comment:
maybe this could be more generic with something like `npmBuildCmd` that is the `npm` script to run? Then if set to `''` or some other sane default don't run anything